clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1540,6 +1540,7 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
   StartFunction(GD, ResTy, Fn, FnInfo, Args, Loc, BodyRange.getBegin());
 
   SyclOptReportHandler &SyclOptReport = CGM.getDiags().getSYCLOptReport();
+  assert(Fn && "Function cannot be null");
   if (SyclOptReport.HasOptReportInfo(FD)) {
```

Comment:
It looks like Fn could be nullptr entering this routine.  There are at least two places above that Fn is checked before it is used, and those checks look necessary to me from the comments.  Further, the routine StartFunction that is called at line 1540 dereferences Fn inside it, so at a minimum moving this assert before that routine is called makes more sense.

I wonder if a better fix may be to change the line below to:
```suggestion
  if (Fn && SyclOptReport.HasOptReportInfo(FD)) {
```
I am not a 100% sure.

Reply:
Thanks @premanandrao for the review. This seems to me a better fix for checking pointer "Fn" inside the routine. 
@elizabethandrews implemented this OptReport, what is opinion for this change? Thank you.

Reply:
I like @premanandrao's suggestion

Reply:
Thank you for the reviews. Yes, this seems to me better fix to check the pointer "Fn" inside the routine. I have updated patch.